registry/storage/tagstore.go

```diff
@@ -112,16 +112,7 @@ func (ts *tagStore) Untag(ctx context.Context, tag string) error {
 		return err
 	}
 
-	if err := ts.blobStore.driver.Delete(ctx, tagPath); err != nil {
-		switch err.(type) {
-		case storagedriver.PathNotFoundError:
-			return nil // Untag is idempotent, we don't care if it didn't exist
```

Comment:
This is no longer true. We must respond with `404 Not Found` if a tag is not found at the API level. So returning the error here and letting the caller decide what to do with it.